CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,120 @@
+# How to Contribute to Jaeger
+
+We'd love your help!
+
+Jaeger is [Apache 2.0 licensed](LICENSE) and accepts contributions via GitHub
+pull requests. This document outlines some of the conventions on development
+workflow, commit message formatting, contact points and other resources to make
+it easier to get your contribution accepted.
+
+We gratefully welcome improvements to documentation as well as to code.
+
+# Certificate of Origin
+
+By contributing to this project you agree to the [Developer Certificate of
+Origin](https://developercertificate.org/) (DCO). This document was created
+by the Linux Kernel community and is a simple statement that you, as a
+contributor, have the legal right to make the contribution. See the [DCO](DCO)
+file for details.
+
+## Making A Change
+
+*Before making any significant changes, please open an
+issue.* Each issue must address the following:
+* Requirement - what kind of business use case are you trying to solve?
+* Problem - what in Jaeger blocks you from solving the requirement?
+* Proposal - what do you suggest to solve the problem or improve the existing situation?
+* Any open questions to address
+
+Discussing your proposed changes ahead of time will make the contribution process smooth for everyone. Once we've discussed your changes and you've made your updates, then open your PR. Each PR must contain:
```

Comment:
* s/Once we've discussed your changes and you've made your updates, then open your PR./Once the approach is agreed upon, make your changes and open a pull request (PR)./
* s/Each PR must contain/Each PR should describe:/